src/app/app.routes.ts

```diff
@@ -6,14 +6,18 @@ import { LoginPageComponent } from './components/login-page/login-page.component
 import { authGuard } from './service/guard/auth.guard'
 import { loginGuard } from './service/guard/login.guard'
 import { RegistrationPageComponent } from './components/registration-page/registration-page.component'
+import { UserListComponent } from './components/user-list/user-list.component'
+import { UserDetailsComponent } from './components/user-details/user-details.component'
 
 export const routes: Routes = [
     { path: "", canActivate: [authGuard], component: TaskComponent, pathMatch: "full" },
-    { path: "create-task", canActivate: [authGuard], component: CreateTaskComponent },
+    { path: "create-task", canActivate: [authGuard], component: CreateTaskComponent, data: { roles: ['ADMIN'] } },
```

Comment:
change it to requiredRoles or something, it was confusing looking at the authGuard with data.roles and route.data['roles']